### PR TITLE
add auto completion for xgboost objective param #2059

### LINKED ISSUE #2059: Add auto completion for xgboost's objective parameter
**Is your feature request related to a problem? Please describe.**
I would appreciate if there is some kind of auto completion for xgboost's objective param, for it has too many options.

**Describe the solution you'd like**
6F666910-8CBF-43A4-AA03-3D7E43E38A06

**Describe alternatives (optional)**
 refer to xgboost doc at https://xgboost.readthedocs.io/en/latest/parameter.html

**Additional Notes**

### PR DESCRIPTION
**Pull request by lhw362950217**
Fix #2059

### CODE REVIEW

File: pkg/sql/codegen/attribute/xgboost_objective_params.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    "rank:ndcg": "Use LambdaMART to perform list-wise ranking where Normalized Discounted Cumulative Gain (NDCG) is maximized",
+    "rank:pairwise": "Use LambdaMART to perform pairwise ranking where the pairwise loss is minimized",
+    "reg:gamma": "gamma regression with log-link. Output is a mean of gamma distribution. It might be useful, e.g., for modeling insurance claims severity, or for any outcome that might be gamma-distributed.",
+    "reg:logistic": "logistic regression",
+    "reg:squarederror": "regression with squared loss.",
+    "reg:squaredlogerror": "regression with squared log loss 1/2[log(pred+1)\u2212log(label+1)]^2",
+    "reg:tweedie": "Tweedie regression with log-link. It might be useful, e.g., for modeling total loss in insurance, or for any outcome that might be Tweedie-distributed.",
+    "survival:cox": "Cox regression for right censored survival time data (negative values are considered right censored). Note that predictions are returned on the hazard ratio scale (i.e., as HR = exp(marginal_prediction) in the proportional hazard function h(t) = h0(t) * HR)."
+}
```
Comment (shendiaomo) [inline pkg/sql/codegen/attribute/xgboost_objective_params.go:34]: Update the `attributeDictionary` in https://github.com/sql-machine-learning/sqlflow/blob/develop/pkg/sql/codegen/xgboost/codegen.go to make sure all the objective works?

Comment (lhw362950217) [reply pkg/sql/codegen/attribute/xgboost_objective_params.go:34]: done

**Review by shendiaomo — DISMISSED** [review]

File: cmd/repl/repl_test.go [hunk 2]
```diff
@@ -618,6 +618,31 @@ func TestComplete(t *testing.T) {
 	c = s.completer(*p.Document())
 	a.Equal(1, len(c))
 	a.Equal("TRAIN", c[0].Text)
+
+	// Test XGBoost objective parameter completion
+	s = newPromptState()
+	p = prompt.NewBuffer()
+	p.InsertText("SELECT * FROM train TO TRAIN xgboost.gbtree WITH objective=", false, true)
+	c = s.completer(*p.Document())
+	a.Equal(15, len(c))
```
Comment (tonyyang-svail) [inline cmd/repl/repl_test.go]: Looks like CI failed on this line
```
    repl_test.go:627: 
        	Error Trace:	repl_test.go:627
        	Error:      	Not equal: 
        	            	expected: 15
        	            	actual  : 14
        	Test:       	TestComplete
```

**Review by tonyyang-svail — COMMENTED** [review]
Thanks for submitting this auto-completion feature.

File: cmd/repl/repl_test.go (hunk 2 above)
Comment (lhw362950217) [reply cmd/repl/repl_test.go]: fixed

**Review by shendiaomo — APPROVED** [review]
LGTM

**Comment by lhw362950217** [comment]
Fix #2059

**Pull request merged**